packages/datadog-esbuild/index.js

```diff
@@ -55,6 +56,18 @@ module.exports.name = 'datadog-esbuild'
 
 module.exports.setup = function (build) {
   const externalModules = new Set(build.initialOptions.external || [])
+
+  if (DD_BUILD_ESM) {
```

Comment:
```suggestion
  if (build.initialOptions.format === 'esm' || build.initialOptions.outfile.endsWith('.mjs')) {
```

The second case would need an additional test case where we do not use `format`.